### ISSUE TITLE
TLDR -  C client does have an environment variable to select cache directory location 

### ISSUE THREAD

**Issue by andrewcrook** [b3nj5m1n/xdg-ninja#295]
``` json
 "help": "Currently tldr-pages do not have any specification for clients on where to cache tldr-pages. See the relevant [issue](https://github.com/tldr-pages/tldr/issues/876) and [specification](https://github.com/tldr-pages/tldr/blob/main/CLIENT-SPECIFICATION.md)\n\nOfficial tldr-nodejs-client, tldr-python-client or tldr-c-client do not support XDG Base Directories.\n\n**Use [Tealdeer](https://github.com/dbrgn/tealdeer), a rust client for tldr that supports XDG Base Directories instead.**\n”
```
[source](https://github.com/b3nj5m1n/xdg-ninja/blob/33b6491dd4561f134a89a22db5513df27219a15e/programs/tldr.json#L7C20-L7C20)

whilst the tldr c client doesnt use XDG it does look for the following environment variable to store the cache 

`TLDR_CACHE_DIR`

This can be used, for example,  to achieve 

``` bash
TLDR_CACHE_DIR=“$XDG_CACHE_HOME/tldrc”
```

``` c
gethome(void)
{
    char const *homedir = NULL;
    struct passwd *uid;

    if ((homedir = getenv("TLDR_CACHE_DIR")) != NULL)
        return homedir;

/* see code for the rest ....  */

    return homedir;
}
```
[source](https://github.com/tldr-pages/tldr-c-client/blob/ff0bd8ad12bbc2cc8f77449a04a3843dc7954691/src/utils.c#L29)

**Comment by flexagoon** [comment]
Is this feature only available in the C client? How can users determine which client they're using?

**Comment by andrewcrook** [comment]
# Is this feature only available in the C client? 

Yes looks like it, despite being in the same GitHub account it looks like there are differences in specification between ports. 
 
**tldr-c-client** - uses` $TLDR_CACHE_DIR`

**tldr-python-client** - already uses XDG_CACHE_HOME  
https://github.com/tldr-pages/tldr-python-client/blob/1ba127a4408376ebc0957590476edf99554b7601/tldr.py#L80

**tldr-Node** - doesn’t support XDG looks hardwired to $HOME/.tldr
https://github.com/tldr-pages/tldr-node-client/issues/346
https://github.com/tldr-pages/tldr-node-client/blob/d38403c0be60898da67ec049ed1241c1f7a9dabe/lib/config.js#L15

**tldrc (The Rust Client)**  - Uses the package `dirs` I believe this to support XDG (however, I am not sure how this packages deals with macOS it is clear enough, that is, if XDG overrides the Apple recommended folders as mention in the docs)
https://crates.io/crates/dirs

#  How can users determine which client they're using?

Good point and I bet there are already examples of this issue?

There are a number of ways to identify scripts and binaries. 
Binary and script type should probably be a new feature for conditionals in the database just like OS probably should be.
